integrations/optimizely/HISTORY.md

```diff
@@ -1,4 +1,13 @@
 
+4.0.0 / 2020-06-12
+==================
+
+  * Don't attempt to interface with Optimizely Classic Web, since Classic is finally dead
+  * Prepare to support Optimizely Edge, an alternative to Optimizely Web
+  * Drop all references to customCampaignProperties. It seems to have been documented [here](https://segment.com/docs/connections/destinations/catalog/optimizely-web/#settings) but it couldn't possibly have worked.
```

Comment:
Beyond this update to the integration code, *Custom Campaign Properties and Custom Experiment Properties should also be removed from the [docs](https://segment.com/docs/connections/destinations/catalog/optimizely-web/#settings) and from the Segment app.* (I didn't highlight Custom Experiment Properties in HISTORY.md since it was [specific to the Optimizely Classic Web product](https://github.com/segmentio/analytics.js-integrations/blob/d85d089177c88163f723b0589f4397784735bab6/integrations/optimizely/lib/index.js#L263-L277), whose code is being removed entirely.)

Full explanation of why this functionality is surely broken:

* The [docs](https://segment.com/docs/connections/destinations/catalog/optimizely-web/#custom-campaign-properties) indicate that Custom Experiment Properties and Custom Campaign Properties could be used to capture experiment- and campaign-specific properties like `experiment_color` and `campaign_name` in Classic Web and X Web respectively. This makes sense, kinda.
* Unfortunately, the integration code indicates that [for Classic Web](https://github.com/segmentio/analytics.js-integrations/blob/d85d089177c88163f723b0589f4397784735bab6/integrations/optimizely/lib/index.js#L273), the Custom Experiment Properties are being read off of the top-level [`window.optimizely.data` object](https://github.com/optimizely/classic-documentation/blob/310e00b81d805e83d4ed4c3f14f9787b339522b3/classic/javascript/reference/index.md#the-data-object), rather than off of an experiment object
* And [for X Web](https://github.com/segmentio/analytics.js-integrations/blob/d85d089177c88163f723b0589f4397784735bab6/integrations/optimizely/lib/index.js#L383), the Custom Campaign Properties are being read off of a non-existent `window.optimizely.newMockData` property. This property is mocked in index.test.js but does not appear in actual X Web snippets.

Reply:
Thanks for the heads-up! I'll work on getting the documentation updated.